src/utils/use-network-connection.ts

```diff
@@ -64,7 +67,10 @@ export const useNetworkConnection = () => {
     const connection = navigator.connection
     const updateConnectionStatus = () => {
       if (navigator.onLine) {
-        if (connection?.effectiveType === '4g') {
+        if (
+          connection?.effectiveType === '4g' &&
+          (connection.type === 'wifi' || connection.type === 'ethernet')
```

Comment:
@varCepheid This doesn't seem to work quite right, when I am on my laptop on wifi, the video doesn't load automatically. `navigator.connection.effectiveType` is `4g` on my computer and `navigator.connection.type` is not defined. https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/type#browser_compatibility. It looks like that property is not supported on most computers (but it is on phones, except iphones). Is there different logic we can use when connection.type is not defined?

Reply:
Simplest fix I see is running only the first condition when `connection.type` is undefined. Any issue with that? It fixes the issue on my computer, but I don't have a good way to test at the moment on a phone.

Reply:
👍 Sure, works for me. But that means that on iphones it will always load by default 😬 I guess that is fine 🤷